src/geometry/euclidean/basic.lean

```diff
@@ -405,7 +407,129 @@ begin
   exact inner_weighted_vsub p h p h
 end
 
-open affine_subspace
+/-- Suppose that `c₁` is equidistant from `p₁` and `p₂`, and the same
+applies to `c₂`.  Then the vector between `c₁` and `c₂` is orthogonal
+to that between `p₁` and `p₂`.  (In two dimensions, this says that the
+diagonals of a kite are orthogonal.) -/
+lemma inner_vsub_vsub_of_dist_eq_of_dist_eq {c₁ c₂ p₁ p₂ : P} (hc₁ : dist p₁ c₁ = dist p₂ c₁)
+  (hc₂ : dist p₁ c₂ = dist p₂ c₂) : inner (c₂ -ᵥ c₁) (p₂ -ᵥ p₁) = 0 :=
+begin
+  have h : inner ((c₂ -ᵥ c₁) + (c₂ -ᵥ c₁)) (p₂ -ᵥ p₁) = 0,
+  { conv_lhs { congr, congr, rw ←vsub_sub_vsub_cancel_right c₂ c₁ p₁,
+               skip, rw ←vsub_sub_vsub_cancel_right c₂ c₁ p₂ },
+    rw [←add_sub_comm, inner_sub_left],
+    conv_lhs { congr, rw ←vsub_sub_vsub_cancel_right p₂ p₁ c₂,
+               skip, rw ←vsub_sub_vsub_cancel_right p₂ p₁ c₁ },
+    rw [dist_comm p₁, dist_comm p₂, dist_eq_norm_vsub V _ p₁,
+        dist_eq_norm_vsub V _ p₂, ←inner_add_sub_eq_zero_iff] at hc₁ hc₂,
+    simp_rw [←neg_vsub_eq_vsub_rev c₁, ←neg_vsub_eq_vsub_rev c₂, sub_neg_eq_add,
+             neg_add_eq_sub, hc₁, hc₂, sub_zero] },
+  simpa [inner_add_left, ←mul_two, (by norm_num : (2 : ℝ) ≠ 0)] using h
+end
+
+/-- The squared distance between points on a line (expressed as a
+multiple of a fixed vector added to a point) and another point,
+expressed as a quadratic. -/
+lemma dist_smul_vadd_square (r : ℝ) (v : V) (p₁ p₂ : P) :
+  dist (r • v +ᵥ p₁) p₂ * dist (r • v +ᵥ p₁) p₂ =
+    inner v v * r * r + 2 * inner v (p₁ -ᵥ p₂) * r + inner (p₁ -ᵥ p₂) (p₁ -ᵥ p₂) :=
+begin
+  rw [dist_eq_norm_vsub V _ p₂, ←inner_self_eq_norm_square, vadd_vsub_assoc, inner_add_add_self,
+      inner_smul_left, inner_smul_left, inner_smul_right],
+  ring
+end
+
+/-- The condition for two points on a line to be equidistant from
+another point. -/
+lemma dist_smul_vadd_eq_dist {v : V} (p₁ p₂ : P) (hv : v ≠ 0) (r : ℝ) :
+  dist (r • v +ᵥ p₁) p₂ = dist p₁ p₂ ↔ (r = 0 ∨ r = -2 * inner v (p₁ -ᵥ p₂) / inner v v) :=
+begin
+  conv_lhs { rw [←mul_self_inj_of_nonneg dist_nonneg dist_nonneg, dist_smul_vadd_square,
+                 ←sub_eq_zero_iff_eq, add_sub_assoc, dist_eq_norm_vsub V p₁ p₂,
+                 ←inner_self_eq_norm_square, sub_self] },
+  have hvi : inner v v ≠ 0, by simpa using hv,
+  have hd : discrim (inner v v) (2 * inner v (p₁ -ᵥ p₂)) 0 =
+    (2 * inner v (p₁ -ᵥ p₂)) * (2 * inner v (p₁ -ᵥ p₂)),
+  { rw discrim, ring },
+  rw [quadratic_eq_zero_iff hvi hd, add_left_neg, zero_div, neg_mul_eq_neg_mul,
+      ←mul_sub_right_distrib, sub_eq_add_neg, ←mul_two, mul_assoc, mul_div_assoc,
+      mul_div_mul_left, mul_div_assoc],
+  norm_num
+end
+
+open affine_subspace finite_dimensional
+
+/-- Distances from two different points determine at most two points
+in a two-dimensional subspace containing those points (two circles
```

Comment:
Perhaps better with the variable names intermingled with the text?
```suggestion
/-- Distances r₁ r₂ of p from two different points c₁ c₂ determine at most two points p₁ p₂
in a two-dimensional subspace containing those points (two circles
```

Reply:
(Variable names should be surrounded with backticks. I would've edited the suggestion above except that it makes the lines longer than 100 characters.)

Reply:
Yeah, you'll need to rewrap the line below this one